test/models/User.ts

```diff
@@ -39,10 +39,14 @@ User.init(
 )
 
 // Hooks
-User.afterFind((users, options: FindOptions) => {
+User.afterFind((users, options) => {
     console.log('found')
 })
 
+// TODO: VSCode shows the typing being correctly narrowed but doesn't do it correctly
```

Comment:
Why is this vscode specific? vscode uses the TS language service

Reply:
I am not sure what is happening either, when I hover over the declaration in VSCode it correctly determines the callback type, but the text pane can't. Can you maybe repro this?